crates/ty_python_semantic/src/types/class.rs

```diff
@@ -2032,9 +2032,10 @@ impl<'db> ClassLiteral<'db> {
     /// class Foo(Bar, metaclass=Baz): ...
     ///       ^^^^^^^^^^^^^^^^^^^^^^^
     /// ```
-    pub(super) fn header_range(self, db: &'db dyn Db, module: &ParsedModuleRef) -> TextRange {
+    pub(super) fn header_range(self, db: &'db dyn Db) -> TextRange {
         let class_scope = self.body_scope(db);
-        let class_node = class_scope.node(db).expect_class(module);
+        let module = parsed_module(db.upcast(), class_scope.file(db)).load(db.upcast());
```

Comment:
I moved the `parsed_module` query inside of `ClassLiteral::header_range` to avoid this footgun in the future, but this likely doesn't apply to all uses of `header_range`, so we could instead make the correct query at call-site. @MichaReiser let me know if you would prefer that, because you mentioned you wanted to remain consistent with non-salsa-queries taking a `&ParsedModuleRef`.